warp/tests/test_sparse.py

```
@@ -972,6 +972,55 @@ def test_bsr_mm_max_new_nnz(test, device):
     test.assertRegex(output, r"exceeded")
 
 
+@wp.kernel
+def _spin_then_bump_offsets(offsets: wp.array[int], mult: wp.array[float], nrow: int, nnz: int, iters: int):
+    """Burn time, then write the block count. Keeps the replayed graph busy
+    long enough that an improperly ordered host readback loses the race.
+    The multiplier comes from memory and the arms differ, so the spin loop
+    cannot be folded away."""
+    acc = mult[0]
+    for i in range(iters):
```

Ruff (0.16.1)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)